datanode/gateway/graphql/referral_set_resolver.go

```diff
@@ -22,33 +22,12 @@ func (r *referralSetRefereeResolver) RefereeID(ctx context.Context, obj *v2.Refe
 	return obj.Referee, nil
 }
 
-type referralSetResolver VegaResolverRoot
-
-func (r *referralSetResolver) Stats(ctx context.Context, obj *v2.ReferralSet, epoch *int, referee *string) (*v2.ReferralSetStats, error) {
-	var atEpoch *uint64
-	if epoch != nil {
-		atEpoch = ptr.From(uint64(*epoch))
-	}
-
-	req := v2.GetReferralSetStatsRequest{
-		ReferralSetId: obj.Id,
-		AtEpoch:       atEpoch,
-		Referee:       referee,
-	}
-
-	res, err := r.tradingDataClientV2.GetReferralSetStats(ctx, &req)
-	if err != nil {
-		return nil, err
-	}
-	return res.Stats, nil
-}
-
 type referralSetStatsResolver VegaResolverRoot
 
-func (r *referralSetStatsResolver) AtEpoch(_ context.Context, obj *v2.ReferralSetStats) (*int, error) {
-	if obj == nil {
-		return nil, nil
+func (r *referralSetStatsResolver) AtEpoch(_ context.Context, obj *v2.ReferralSetStats) (int, error) {
+	if obj.AtEpoch > math.MaxInt {
```

Comment:
I'm not really sure why we'd take a `uint64` and restrict it to the max value of an int here... why is a `uint64` not possible, or why not use a supported/larger type that can hold all _valid_ values?

Reply:
AFAIK, GraphQL only support integer out of the box. For ref:
>The GraphQL schema language supports the scalar types of String, Int, Float, Boolean, and ID, so you can use these directly in the schema you pass to buildSchema.
> \- [GraphQL documentation](https://graphql.org/graphql-js/basic-types/)

About _holding all valid values_, I think we only have strings, out of the box. Otherwise, it down to creating a specific GraphQL object with a specific resolver.

Reply:
yea  this is fine, not uint in gql, only Ints, and in this case it will be good enough.